packages/nimble-components/src/menu/tests/menu.spec.ts

```diff
@@ -8,4 +15,121 @@ describe('Menu', () => {
+                    </${menuTag}>
+                `);
+            }
+            let element: Menu;
+            let item1: Element;
+            let item2: Element;
+            let connect: () => Promise<void>;
+            let disconnect: () => Promise<void>;
+
+            beforeEach(async () => {
+                ({ element, connect, disconnect } = await setup());
+                await connect();
+                await waitForUpdatesAsync();
+                const items = element.querySelectorAll('[role="menuitem"]');
+                item1 = items[0]!;
+                item2 = items[1]!;
+            });
+
+            afterEach(async () => {
+                await disconnect();
+            });
+
+            it('indents items when an item with an icon is slotted after connecting', async () => {
+                expect(item1.classList.contains('indent-0')).toBeTrue();
+                expect(item2.classList.contains('indent-0')).toBeTrue();
+
+                const item3 = document.createElement(name);
+                const icon = document.createElement(iconCheckTag);
+                icon.slot = 'start';
+                item3.appendChild(icon);
+                element.appendChild(item3);
+                await waitForUpdatesAsync();
+
+                expect(item1.classList.contains('indent-1')).toBeTrue();
+                expect(item2.classList.contains('indent-1')).toBeTrue();
+                expect(item3.classList.contains('indent-1')).toBeTrue();
+            });
+
+            it('does not indent items when the only item with an icon is removed', async () => {
+                const icon = document.createElement(iconCheckTag);
+                icon.slot = 'start';
+                item1.appendChild(icon);
+                await waitForUpdatesAsync();
+
+                element.removeChild(item1);
+                await waitForUpdatesAsync();
+
+                expect(item2.classList.contains('indent-0')).toBeTrue();
+                expect(item2.classList.contains('indent-1')).toBeFalse();
+            });
+
+            it('indents newly added item (without icon) when there was already an item with an icon', async () => {
+                const icon = document.createElement(iconCheckTag);
+                icon.slot = 'start';
+                item1.appendChild(icon);
+                await waitForUpdatesAsync();
+
+                const item3 = document.createElement(name);
+                element.appendChild(item3);
+                await waitForUpdatesAsync();
+
+                expect(item3.classList.contains('indent-1')).toBeTrue();
+            });
+
+            it('does not indent newly added item (without icon) when there were no items with icons', async () => {
+                const item3 = document.createElement(name);
+                element.appendChild(item3);
+                await waitForUpdatesAsync();
+
+                expect(item3.classList.contains('indent-0')).toBeTrue();
+            });
```

Comment:
Seems like a weird test case, but I discovered that FAST's menu item code [apparently defaults the indentation level to 1](https://github.com/microsoft/fast/blob/1f9c97df9acae24be44f239bd66b5b2157c7f3e5/packages/web-components/fast-foundation/src/menu-item/menu-item.ts#L169) instead of 0, for some reason, so this can legitimately fail if we're not calling `Menu.setItems()` when new items are added.